src/Front/Ast.hs

```diff
@@ -1,45 +1,31 @@
 module Front.Ast where
 
+import           Common.Identifiers             ( Identifiable(..)
+                                                , Identifier
+                                                )
 import           Common.Pretty
-import           Data.List                      ( intersperse )
-
--- | A type variable name (e.g., a, b)
-type TVarId = String
-
--- | A type constructor name (e.g., Int, Bool)
-type TConId = String
-
--- | A data constructor name (e.g., Just, False)
-type DConId = String
 
--- | A type class name (e.g., Eq, Ord)
-type TClassId = String
-
--- | A variable name (e.g., x, y, f)
-type VarId = String
-
--- | An operator name (e.g., +, foo)
-type OperatorId = String
+import           Data.List                      ( intersperse )
```

Comment:
There's no point in creating all these type aliases here when these aren't even being differentiated by the parser. Instead, we use `Identifier` for any such node in the AST.